addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/CacheController.java

```diff
@@ -861,20 +861,15 @@ private boolean checkAlwaysMiss(String url, Method method, Cache cache) {
             String indicValue = msg.getResponseHeader().getHeader(cache.getIndicator());
             if (indicValue == null || indicValue.isEmpty()) {
                 return true;
-            } else {
-                if (!this.checkCacheHit(indicValue, cache) && cache.getIndicator() != null) {
-                    sleeper(2000);
-                    httpSender.sendAndReceive(msg);
-                    addCacheMessage(msg);
-                    indicValue = msg.getResponseHeader().getHeader(cache.getIndicator());
-                    if (this.checkCacheHit(indicValue, cache)) {
-                        return false;
-                    } else {
-                        return true;
-                    }
-                }
-                return false;
             }
+            if (!this.checkCacheHit(indicValue, cache) && cache.getIndicator() != null) {
+                sleeper(2000);
+                httpSender.sendAndReceive(msg);
+                addCacheMessage(msg);
+                indicValue = msg.getResponseHeader().getHeader(cache.getIndicator());
+                return this.checkCacheHit(indicValue, cache);
```

Comment:
It was returning false when true, should have test.

Reply:
Will review, and see what I can do/add. I did double check that existing tests passed.

Reply:
Agreed the logic was the other way around. However, I can't come up with a test case where it matters 😞

Reply:
You mean based on the existing tests?

Reply:
Yes, shouldNotFindCacheWithAnyCacheBuster seems to handle one case. But, really passes whether or not that block is true or false 🤷‍♂️

You can validate simply with:
```java
                System.out.println("Here");
                return false;//!this.checkCacheHit(indicValue, cache);
```
and
```java
                System.out.println("Here");
                return true;//!this.checkCacheHit(indicValue, cache);
```

Reply:
Any thoughts?

Reply:
I will have to check the code/tests.